app/controllers/storytime/posts_controller.rb

```diff
@@ -10,11 +13,17 @@ def index
     def show
       params[:id] = params[:id].split("/").last
 
-      @post = if params[:preview]
-        Post.find_preview(params[:id])
-      else
-        Post.published.friendly.find(params[:id])
+      @post = begin
+        if params[:preview]
+          Post.find_preview(params[:id])
+        else
+          Post.published.friendly.find(params[:id])
+        end
+      rescue ActiveRecord::RecordNotFound
+        nil
       end
+      
+      return super if @post.nil?
```

Comment:
I'd vote for us not to expect an exception in a normal execution path (exceptions are expensive). How about:
```
return super unless Post.exists? params[:id]

@post = if params[:preview]
  Post.find_preview(params[:id])
else
  Post.published.friendly.find(params[:id])
end
```